src/components/TimeseriesDataExport/TimeseriesDataExport.tsx

```diff
@@ -118,11 +87,65 @@ const TimeseriesDataExportFC: FC<TimeseriesDataExportFormProps> = (
     imageDownloadBtn,
   } = lang;
 
+  const getLimits = async (request: DatapointsMultiQuery) => {
+    let { start = 0, end = 0 } = request;
+
+    const project = context!.project || '';
+    const url = `/api/v1/projects/${project}/timeseries/data/latest`;
+    // @ts-ignore
+    const items = request.items.map(({ id }) => ({ id }));
+
+    const endResults = await context!.post<{ items: any[] }>(url, {
+      data: { items },
+    });
+    end = Math.min(
+      ...endResults.data.items.map(item => +item.datapoints[0].timestamp, end)
+    );
+
+    const getFirstDatapointRequest = {
+      ...request,
+      start,
+      end,
+      limit: CELL_LIMIT,
+    };
+
+    const startResult = await context!.datapoints.retrieve(
+      getFirstDatapointRequest
+    );
+    start = Math.max(
+      // @ts-ignore
+      ...startResult.map(({ datapoints }) => +datapoints[0].timestamp),
+      // @ts-ignore
+      start
+    );
+
+    return { start, end };
+  };
+
+  const fetchDataPoints = async (request: DatapointsMultiQuery) => {
+    const { start = 0, end = 0 } = await getLimits(request);
```

Comment:
what's the purpose to have a more accurate start-end value?

Reply:
It was implemented after being recommended by @sanderland. Users will often specify 0 or very low start-values The old implementation would then make a significant amount of pointless calls.